packages/plugin-ext/src/main/browser/webviews-main.ts

```diff
@@ -107,15 +161,28 @@ export class WebviewsMainImpl implements WebviewsMain {
         }
     }
     $reveal(handle: string, showOptions: WebviewPanelShowOptions): void {
-        const webview = this.getWebview(handle);
-        if (webview.isDisposed) {
+        const view = this.getWebview(handle);
+        if (view.isDisposed) {
             return;
         }
-        // FIXME handle view column here too!
+        if (showOptions.viewColumn !== undefined || showOptions.area !== undefined) {
+            this.viewColumnService.updateViewColumns();
+            const options = this.viewsOptions.get(view.id);
+            if (!options) {
+                return;
+            }
+            const columnIds = showOptions.viewColumn ? this.viewColumnService.getViewColumnIds(showOptions.viewColumn) : [];
+            if (columnIds.indexOf(view.id) === -1 || options.panelOptions.area !== showOptions.area) {
+                this.addOrReattachWidget(options.panelId, showOptions);
+                this.handler.cancel();
```

Comment:
Would it work with multiple webviews? As far as I understood `handler` updates all view options and here you want to cancel an update only for one, but cancel for all instead.

Maybe it would be better to have debounce function per a view id.

Reply:
fixed

Reply:
fixed